# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_border_solid.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_border_solid.glsl
--- a/gfx/wr/webrender/res/cs_border_solid.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_border_solid.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -10,34 +10,34 @@
 
 // For edges, the colors are the same. For corners, these
 // are the colors of each edge making up the corner.
-flat varying vec4 vColor0;
-flat varying vec4 vColor1;
+flat varying mediump vec4 vColor0;
+flat varying mediump vec4 vColor1;
 
 // A point + tangent defining the line where the edge
 // transition occurs. Used for corners only.
-flat varying vec4 vColorLine;
+flat varying mediump vec4 vColorLine;
 
 // A boolean indicating that we should be mixing between edge colors.
 // Packed in to a vector to work around bug 1630356.
-flat varying ivec2 vMixColors;
+flat varying mediump ivec2 vMixColors;
 
 // xy = Local space position of the clip center.
 // zw = Scale the rect origin by this to get the outer
 // corner from the segment rectangle.
-flat varying vec4 vClipCenter_Sign;
+flat varying highp vec4 vClipCenter_Sign;
 
 // An outer and inner elliptical radii for border
 // corner clipping.
-flat varying vec4 vClipRadii;
+flat varying highp vec4 vClipRadii;
 
 // Position, scale, and radii of horizontally and vertically adjacent corner clips.
-flat varying vec4 vHorizontalClipCenter_Sign;
-flat varying vec2 vHorizontalClipRadii;
-flat varying vec4 vVerticalClipCenter_Sign;
-flat varying vec2 vVerticalClipRadii;
+flat varying highp vec4 vHorizontalClipCenter_Sign;
+flat varying highp vec2 vHorizontalClipRadii;
+flat varying highp vec4 vVerticalClipCenter_Sign;
+flat varying highp vec2 vVerticalClipRadii;
 
 // Local space position
-varying vec2 vPos;
+varying highp vec2 vPos;
 
 #define SEGMENT_TOP_LEFT        0
 #define SEGMENT_TOP_RIGHT       1